internal/reference/reference.go

```diff
@@ -109,6 +110,31 @@ func NewResolver(input string) (Resolver, error) {
 	}, nil
 }
 
+func cleanInput(input string) string {
+	replacePrefix := func(s string, replacements map[string]string) string {
+		for prefix, replacement := range replacements {
+			if strings.HasPrefix(s, prefix) {
+				return replacement + s[len(prefix):]
+			}
+		}
+		return s
+	}
+	input = replacePrefix(input, map[string]string{
+		".payload":   ".Payload",
+		".key":       ".Key",
+		".metadata":  ".Metadata",
+		".position":  ".Position",
+		".operation": ".Operation",
+	})
+	if strings.HasPrefix(input, ".Payload") {
+		input = replacePrefix(input, map[string]string{
+			".Payload.before": ".Payload.Before",
+			".Payload.after":  ".Payload.After",
+		})
+	}
```

Comment:
@lovromazgon  any reason you chose to do this rather than include the options above?

Reply:
Because iterating maps in Go is random and we could hit the key `.payload` before we hit `.payload.before` for instance, which would then only rename the first part and result in `.Payload.before`. Notice that `replacePrefix` returns after it finds the first replacement. Another approach would be to make `replacePrefix` execute all replacements it finds, then we could simplify the call. Maybe that would be simpler, let me try.

Reply:
We went down the rabbit hole and discussed in private. Long story short, it won't work, we'll leave it as it is 😄